v2/pkg/nodenet/pod.go

```diff
@@ -422,12 +454,21 @@ func (pn *podNetwork) Update(podIPv4, podIPv6 net.IP, hook SetupHook) error {
 
 	var netNsPath string
 	for _, c := range podConfigs {
-		// When both c.IPvX and podIPvX are nil, net.IP.Equal() returns always true.
-		// To avoid comparing nil to nil, confirm c.IPvX is not nil.
-		if (c.IPv4 != nil && c.IPv4.Equal(podIPv4)) || (c.IPv6 != nil && c.IPv6.Equal(podIPv6)) {
-			netNsPath, err = getNetNsPath(c.HostVethName)
-			if err != nil {
-				return err
+		if pn.enableIPAM {
+			// When both c.IPvX and podIPvX are nil, net.IP.Equal() returns always true.
+			// To avoid comparing nil to nil, confirm c.IPvX is not nil.
+			if (c.IPv4 != nil && c.IPv4.Equal(podIPv4)) || (c.IPv6 != nil && c.IPv6.Equal(podIPv6)) {
+				netNsPath, err = getNetNsPath(c.HostVethName)
+				if err != nil {
+					return err
+				}
+			}
+		} else {
+			if c.ContainerId == string(pod.UID) {
```

Comment:
I've added comment:

```
// We need to bind PodNetConf with Pod. When Coil is a primary CNI IP address can be used for that.
// However different CNIs manage IP addresses differently, therefore, in egress-only mode instead of
// storing container's ID in ContainerId, we store pod's UID there to be able to identify pod that was
// used to create particular PodNetConf.
```

Please let me know if this is what you were thinking about.

Reply:
Thank you.
This is what I want!